sdk-api/src/main/java/dev/restate/sdk/SharedWorkflowContext.java

```diff
@@ -10,8 +10,34 @@
 
 import dev.restate.sdk.common.DurablePromiseKey;
 
+/**
+ * This interface extends {@link Context} adding access to the workflow instance key-value state
```

Comment:
Can we first say what it is and when to use it? 
Maybe tune it a bit:
`This interface should be used as the context for the shared handlers of workflow services`